cornice/util.py

```diff
@@ -128,7 +128,13 @@ def extract_json_data(request):
     if request.body:
         try:
             body = simplejson.loads(request.body)
-            return body
+            if isinstance(body, dict):
```

Comment:
Then maybe we could add a safety check in Service asserting that the given colander schema inherits from colander.MappingSchema ?